### PR TITLE
Add mock for catchError

### PR DESCRIPTION
**Pull request by nre-ableton**
ptal @AbletonDevTools/gotham-city, thanks!

### CODE REVIEW

**Comment by ala-ableton** [comment]
@nre-ableton please rebase if you want me to review, thanks!

**Comment by nre-ableton** [comment]
@ala-ableton Done, and also added the appropriate cleanups/tests.

File: src/test/groovy/com/ableton/JenkinsMocksTest.groovy [hunk 1]
```diff
@@ -6,13 +6,21 @@ import static org.junit.Assert.assertTrue
...
 import org.junit.After
+import org.junit.Before
 import org.junit.Test
 
 
 /**
  * Tests for the JenkinsMocks class.
  */
 class JenkinsMocksTest extends BasePipelineTest {
+  @SuppressWarnings('UnnecessaryOverridingMethod')
```
Comment (ala-ableton) [inline src/test/groovy/com/ableton/JenkinsMocksTest.groovy]: I agree with CodeNarc, why are we overriding this method?

Comment (nre-ableton) [reply src/test/groovy/com/ableton/JenkinsMocksTest.groovy]: For reasons that I don't entirely understand, without this override the `currentBuild` variable is not correctly bound and the tests fail with this error:

```
Caused by: groovy.lang.MissingPropertyException: No such property: currentBuild for class: groovy.lang.Binding
	at groovy.lang.Binding.getVariable(Binding.java:58)
	at groovy.lang.Binding$getVariable$0.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
```

Comment (ala-ableton) [reply src/test/groovy/com/ableton/JenkinsMocksTest.groovy]: Could you paste a link to the failing Jenkins job? I would like to see the whole thing in context.

Comment (nre-ableton) [reply src/test/groovy/com/ableton/JenkinsMocksTest.groovy]: @ala-ableton I only ever saw the error locally. However I just pushed a WIP commit with should also fail on Jenkins (hopefully 😅 ).

Comment (nre-ableton) [reply src/test/groovy/com/ableton/JenkinsMocksTest.groovy]: @ala-ableton Ok, it just failed on Jenkins as well: http://newci.office.ableton.com:8080/blue/organizations/jenkins/AbletonDevTools-jenkins-pipeline-mocks/detail/AbletonDevTools-jenkins-pipeline-mocks/40/tests

**Comment by nre-ableton** [comment]
So, to summarize a discussion we had some days ago on Slack, the reason that the tests fail without the `setUp()` override is that the super's `setUp()` method is not annotated with `@Before`. I forked JenkinsPipelineUnit and played around with the code in my IDE, and discovered that the reason it probably lacks this annotation is that JUnit is defined in their `build.gradle` file as a `testCompile` dependency, meaning that the unit tests depend on it but the framework itself does not. Hence, `import org.junit.*` would cause a compile error when done in the non-test sources.

I'm guessing that the reason for doing this is to not require a hardcoded dependency between JenkinsPipelineUnit and JUnit. However, I made a GitHub issue here to raise the question:

https://github.com/jenkinsci/JenkinsPipelineUnit/issues/90

IMHO we should revert the WIP commit and just live with the empty override and suppression.

**Comment by nre-ableton** [comment]
@ala-ableton I've dropped the WIP commit on this branch and added a comment explaining why the empty override is necessary.

File: src/test/groovy/com/ableton/JenkinsMocksTest.groovy [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
 import org.junit.After
+import org.junit.Before
 import org.junit.Test
 
 
 /**
  * Tests for the JenkinsMocks class.
  */
 class JenkinsMocksTest extends BasePipelineTest {
+  // The super class is not annotated with @Before since JenkinsPipelineUnit does not
```
Comment (ala-ableton) [inline src/test/groovy/com/ableton/JenkinsMocksTest.groovy]: "The super class is not annotated with @Before" -> "The overridden method in not annotated with @Before", or "The method in the super class is not annotated with @Before".

Comment (nre-ableton) [reply src/test/groovy/com/ableton/JenkinsMocksTest.groovy]: True, that's not very well-worded. Amended.

**Comment by ala-ableton** [comment]
LGTM :+1:

**Pull request merged**